precompiles/werc20/tx.go

```diff
@@ -0,0 +1,84 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package werc20
+
+import (
+	"fmt"
+	"math/big"
+
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	cmn "github.com/evmos/evmos/v20/precompiles/common"
+	"github.com/evmos/evmos/v20/x/evm/core/vm"
+	evmtypes "github.com/evmos/evmos/v20/x/evm/types"
+)
+
+const (
+	// DepositMethod defines the ABI method name for the IWERC20 deposit
+	// transaction.
+	DepositMethod = "deposit"
+	// WithdrawMethod defines the ABI method name for the IWERC20 withdraw
+	// transaction.
+	WithdrawMethod = "withdraw"
+)
+
+// Deposit handles the payable deposit function. It retrieves the deposited amount
+// and sends it back to the sender using the bank keeper.
+func (p Precompile) Deposit(ctx sdk.Context, contract *vm.Contract, stateDB vm.StateDB) ([]byte, error) {
```

Comment:
with STRv2, shouldn't this be a no-op too?
cause all the EVMOS (native coin balance) will desplay as WEVMOS balance

Reply:
But what if we just emit the event? 'cause the deposit function is payable on the solidity interface, but then on the code that is executed when calling the function, should be a no-op. Or I'm missing smth?